mailbox/plugin/quota-mailing-cassandra/src/main/java/org/apache/james/mailbox/quota/cassandra/CassandraQuotaThresholdHistoryStoreModule.java

```diff
@@ -0,0 +1,71 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.quota.cassandra;
+
+import java.util.List;
+
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.components.CassandraTable;
+import org.apache.james.backends.cassandra.components.CassandraType;
+import org.apache.james.backends.cassandra.utils.CassandraConstants;
+
+import com.datastax.driver.core.DataType;
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+import com.google.common.collect.ImmutableList;
+
+public class CassandraQuotaThresholdHistoryStoreModule implements CassandraModule {
+    private final List<CassandraTable> tables;
+    private final List<CassandraType> types;
+
+    public CassandraQuotaThresholdHistoryStoreModule() {
+        tables = ImmutableList.of(
+            new CassandraTable(CassandraQuotaThresholdHistoryStoreTable.SIZE_TABLE_NAME,
+                SchemaBuilder.createTable(CassandraQuotaThresholdHistoryStoreTable.SIZE_TABLE_NAME)
+                    .ifNotExists()
+                    .addPartitionKey(CassandraQuotaThresholdHistoryStoreTable.USER, DataType.varchar())
+                    .addClusteringColumn(CassandraQuotaThresholdHistoryStoreTable.THRESHOLD, DataType.cdouble())
```

Comment:
I would have put instant before threshold... Ordering by time would be more natural.

Reply:
I first did as you tell, but then I switch because I think it could be more interesting to filter on (user, threshold) than on (user, instant). WDYT?